test/drenv/__main__.py

```diff
@@ -134,10 +132,10 @@ def func(profile, **options):
                 f.result()
             except Exception:
                 logging.exception("[%s] Cluster failed", futures[f])
-                failed = True
-
-    if failed:
-        sys.exit(1)
+                # This is not the great way to terminate since we may leave
+                # running addons scripts, but it is better than running for
+                # many minutes after a failure.
+                os._exit(1)
```

Comment:
Better to let the error propagate to the caller, and handle the error in a top level error handler.

If this is called in the main executor used in cmd_start(), this will propagate to main(). If is called in the secondary executor called from start_cluster, it will propagate back to cmd_start() and then to main().